src/lib/createTchapRoom.ts

```diff
@@ -23,41 +19,61 @@ export default class TchapCreateRoom {
         tchapRoomType: TchapRoomType,
         federate: boolean = DEFAULT_FEDERATE_VALUE): IOpts {
         const opts: IOpts = {};
-        const createRoomOpts: ITchapCreateRoomOpts = {};
+        const createRoomOpts: ICreateRoomOpts = {};
         opts.createOpts = createRoomOpts;
 
         //tchap common options
         createRoomOpts.name = name;
         opts.guestAccess = false; //guest access are not authorized in tchap
 
         createRoomOpts.creation_content = { 'm.federate': federate };
+        createRoomOpts.initial_state = createRoomOpts.initial_state || [];
 
         switch (tchapRoomType) {
             case TchapRoomType.Forum: {
             //"Forum" only for tchap members and not encrypted
-                createRoomOpts.accessRule = TchapRoomAccessRule.Restricted;
                 createRoomOpts.visibility = Visibility.Public;
                 createRoomOpts.preset = Preset.PublicChat;
+                createRoomOpts.initial_state.push({
```

Comment:
```suggestion
                // Here we could have used createRoomOpts.accessRule directly, but since accessRules are a custom Tchap event, it is ignored by later code. So we use createRoomOpts.initial_state, which works properly.
                createRoomOpts.initial_state.push({
```